### DISCUSSION TITLE
Thread Block Scheduler Policy

### DISCUSSION THREAD

**Discussion by michaelzhiluo** [NVIDIA/open-gpu-kernel-modules#346] [General]
I was looking into modifying the scheduler policy for the thread block scheduler. There seems to be prior papers (i.e. [this](https://cake.wpi.edu/assets/papers/gilman20_performance.pdf)) that have studied the behavior of the current scheduling policies. However, given the partial open-source release, it might be possible to directly observe the policy now. I was wondering where the code for the scheduler is located, or it is still proprietary (deeper into firmware or ...)

**Comment by arisu3** [comment]
This is most likely implemented in the GSP (which runs proprietary firmware) or in the silicon itself, not the driver.
